File: core/src/rustyice/game/actors/TestActor.java
package rustyice.game.actors;

import com.badlogic.gdx.graphics.g2d.Sprite;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.graphics.g2d.TextureAtlas;

import rustyice.game.physics.components.SBPhysicsComponent;

public class TestActor extends Actor {

    private transient Sprite testSprite;
    private SBPhysicsComponent pComponent;

    public TestActor() {
        super();
        setPhysicsComponent(pComponent = new SBPhysicsComponent(this));
        //pComponent.setFlying(true);
        testSprite = new Sprite();

        setPosition(5, 5);
        setWidth(1.9f);
        setHeight(1.9f);
    }

    @Override
    public void update(float delta) {
        super.update(delta);
        // walk(0,1, .95f);
    }

    @Override
    public SBPhysicsComponent getPhysicsComponent() {
        return this.pComponent;
    }

    @Override
    public void render(SpriteBatch batch) {
        this.testSprite.setX(getX() - getWidth() / 2);
        this.testSprite.setY(getY() - getHeight() / 2);
        this.testSprite.setRotation(getRotation());
        this.testSprite.draw(batch);
    }

    @Override
    public void init() {
        super.init();
        this.pComponent.addRectangle(getWidth(), getHeight(), 2, false);
        this.testSprite = getResources().get("art.atlas", TextureAtlas.class).createSprite("may2015-3");
        this.testSprite.setSize(getWidth(), getHeight());
        this.testSprite.setOrigin(getWidth() / 2, getHeight() / 2);
    }
}

File: core/src/rustyice/game/actors/TestActor.kt
package rustyice.game.actors

import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.graphics.g2d.Sprite
import com.badlogic.gdx.physics.box2d.FixtureDef
import rustyengine.RustyEngine
import rustyice.game.Actor
import rustyice.game.physics.SBPhysicsComponent
import rustyice.graphics.Camera
import rustyice.graphics.RenderLayer
import rustyice.physics.LARGE
import rustyice.physics.WALL

/**
 * @author gabek
 */
class TestActor() : Actor() {
    @Transient private var testSprite: Sprite? = null

    val sbPhysicsComponent: SBPhysicsComponent

    override fun update(delta: Float) {
        super.update(delta)
    }

    override fun render(batch: Batch, camera: Camera, layer: RenderLayer) {
        val testSprite = testSprite
        if(testSprite != null){
            testSprite.x = x - width / 2
            testSprite.y = y - height / 2
            testSprite.rotation = rotation
            testSprite.draw(batch)
        }
    }

    override fun init() {
        super.init()

        val fixtureDef = FixtureDef()
        fixtureDef.density = 1f
        fixtureDef.filter.categoryBits = LARGE.toShort()
        fixtureDef.filter.maskBits = (LARGE or
                                      WALL).toShort()
        sbPhysicsComponent.addRectangle(width, height, fixtureDef)


        val testSprite = Sprite(RustyEngine.resorces.box)
        this.testSprite = testSprite

        testSprite.setSize(width, height)
        testSprite.setOrigin(width/2, height/2)
    }

    init {
        sbPhysicsComponent = SBPhysicsComponent()
        physicsComponent = sbPhysicsComponent
        width = 1.9f
        height = 1.9f
    }
}
